sftp.c

```diff
@@ -847,10 +919,31 @@ do_ls_dir(struct sftp_conn *conn, const char *path,
 				    (lflag & LS_SI_UNITS));
 				mprintf("%s\n", lname);
 				free(lname);
+#if IRONCORE
+			} else {
+				/*  If file has .iron extension, prefix with lock icon  */
+				int offset = iron_extension_offset(fname);
+				char * fname_start = strstr(d[n]->longname, fname);
+				if (fname_start == NULL) {
+					fname_start = d[n]->longname + strlen(d[n]->longname);
+				}
+
+				int len = fname_start - d[n]->longname;
+				mprintf("%-*.*s", len, len, d[n]->longname);
+				char * icon_prefix = (offset > 0) ? IRON_LOCK_ICON : "  ";
```

Comment:
Should all be consistently `>0` now, except for one case which is commented to explain difference.